File: HRO/CMITDW Taal der Wiskunde/o65.java
// opdracht 6.5; Priemgetallen; Zeef van Eratosthenes
public class o65
{
	public static void main(String[] args)
	{
		int i = 0; // index of the prime number
		boolean[] b = new boolean[(int) 1e6]; // double -> int
		
		for (int p=2; (p<b.length); p++) // number to be checked
		{
			if (!b[p])
			{
				i++;
				
				if (p <= 100 || p >= 999900) // skip bulk
					System.out.println(i+"\t"+p);
				
				for (int j=p*2; j<b.length; j+=p)
					b[j] = true;
			}
		}
	}
}

File: HRO/CMITDW Taal der Wiskunde/Lua/o65.lua
-- opdracht 6.5; Priemgetallen; Zeef van Eratosthenes
local t = {}
local n = 1e6
local i = 0 -- index of the prime number

for p = 2, n do -- number to be checked
	if not t[p] then
		i = i + 1
		if p <= 100 or p >= 999900 then -- skip bulk
			print(i, p)
		end
		
		for j = p*2, n, p do
			t[j] = true
		end
	end
end
